nemo/collections/asr/modules/rnnt.py

```diff
@@ -50,6 +50,525 @@
 from nemo.utils import logging
 
 
+class StatelessNet(torch.nn.Module):
+    def __init__(self, context_size, vocab_size, emb_dim, blank_idx, blank_as_pad, normalization_mode, dropout):
+        super().__init__()
+        self.context_size = context_size
+        self.vocab_size = vocab_size
+        self.emb_dim = emb_dim
+        self.dropout = torch.nn.Dropout(dropout)
+        self.norm = torch.nn.Identity()
+        if normalization_mode == 'layer':
+            self.norm = torch.nn.LayerNorm(emb_dim, elementwise_affine=False)
+
+        assert(blank_as_pad)  # So far it only supports when it is set True
+        embeds = []
+        for i in range(self.context_size):
+            # We use different embedding matrices for different context positions.
+            # In this list, a smaller index means more recent history word.
+            # We assign more dimensions for the most recent word in the history.
+            if i != 0:
+                embed_size = emb_dim // 2 // self.context_size
+            else:
+                embed_size = emb_dim - (emb_dim // 2 // self.context_size) * (self.context_size - 1)
```

Comment:
Describe how you are calculating this, step by step.

Reply:
Done.